pkg/kubelet/lifecycle/handlers.go

```diff
@@ -117,7 +121,19 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 			return "", fmt.Errorf("failed to find networking container: %v", status)
 		}
 		host = status.IPs[0]
+		podIP = host
 	}
+
+	if utilfeature.DefaultFeatureGate.Enabled(features.ConsistentHTTPGetHandlers) {
+		req, err := httpprobe.NewRequestForHTTPGetAction(handler.HTTPGet, container, podIP, "lifecycle")
+		if err != nil {
+			return "", err
+		}
+		resp, err := hr.httpDoer.Do(req)
+		return getHTTPRespBody(resp), err
+	}
+
+	// Deprecated code path.
 	var port int
 	if handler.HTTPGet.Port.Type == intstr.String && len(handler.HTTPGet.Port.StrVal) == 0 {
 		port = 80
```

Comment:
No action required, but fwiw, I don't think API validation of lifecycle handlers actually allowed this path to be taken: the port field looks like it is required to be present and a valid port number or a non-empty string:

https://github.com/kubernetes/kubernetes/blob/b6d89e756a706c8179f2829d612e1b24e68e9374/pkg/apis/core/validation/validation.go#L2908-L2913

https://github.com/kubernetes/kubernetes/blob/b6d89e756a706c8179f2829d612e1b24e68e9374/pkg/apis/core/validation/validation.go#L2876-L2881

https://github.com/kubernetes/kubernetes/blob/b6d89e756a706c8179f2829d612e1b24e68e9374/pkg/apis/core/validation/validation.go#L2826-L2831

https://github.com/kubernetes/kubernetes/blob/b6d89e756a706c8179f2829d612e1b24e68e9374/pkg/apis/core/validation/validation.go#L2843-L2855